File: src/stickysituation/stickysituation.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"sort"
	"strconv"
	"strings"
)

// https://open.kattis.com/problems/stickysituation

var sticks []int

func main() {
	io, _ := ioutil.ReadAll(os.Stdin)
	line := strings.Fields(string(io))
	line = line[1:]

	sticks = make([]int, len(line))
	for i, s := range line {
		sticks[i], _ = strconv.Atoi(s)
	}

	sort.Slice(sticks, func(i, j int) bool {
		return sticks[i] < sticks[j]
	})

	// eliminate small sticks (impossible triangle). Only possible if sorted.
	start := 0
	for i := 0; i < len(sticks) - 2; i++ {
		if sticks[i] + sticks[i+1] <= sticks[i+2] {
			start++
		} else {
			break
		}
	}

	sticks = sticks[start:]
	if len(sticks) >= 3 {
		if sticks[0] + sticks[1] > sticks[2] {
			fmt.Println("possible")
		} else {
			fmt.Println("impossible")
		}
	} else {
		fmt.Println("impossible")
	}
}

File: src/stickysituation/StickySituation.java
package stickysituation;

import lib.io.FastReader;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

// https://open.kattis.com/problems/stickysituation

public class StickySituation {

    public static void main(String[] args) {
        FastReader io = new FastReader(System.in);
        io.next();  // skip number of sticks

        String line = io.nextLine();
        String[] stick = line.split("\\s");

        List<Long> sticks = new ArrayList<>();
        for (String s : stick) {
            sticks.add(Long.parseLong(s));
        }

        Collections.sort(sticks);

        int start = 0;
        for (int i = 0; i < sticks.size() - 2; i++) {
            if (sticks.get(i) + sticks.get(i + 1) <= sticks.get(i + 2)) {
                start++;
            } else {
                break;
            }
        }

        if (sticks.size() - 1 - start < 2) {
            System.out.println("impossible");
            return;
        }
        if (sticks.get(start) + sticks.get(start + 1) > sticks.get(start + 2)) {
            System.out.println("possible");
        } else {
            System.out.println("impossible");
        }
    }
}
